agent/agentcontainers/containers_dockercli.go

```diff
@@ -31,6 +33,172 @@ func NewDocker(execer agentexec.Execer) Lister {
 	}
 }
 
+// ContainerEnvInfoer is an implementation of agentssh.EnvInfoer that returns
+// information about a container.
+type ContainerEnvInfoer struct {
+	container string
+	user      *user.User
+	userShell string
+	env       []string
+}
+
+// EnvInfo returns information about the environment of a container.
+func EnvInfo(ctx context.Context, execer agentexec.Execer, container, containerUser string) (*ContainerEnvInfoer, error) {
+	var dei ContainerEnvInfoer
+	dei.container = container
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	if containerUser == "" {
+		// Get the "default" user of the container if no user is specified.
+		// TODO: handle different container runtimes.
+		cmd, args := WrapDockerExec(container, "")("whoami")
+		execCmd := execer.CommandContext(ctx, cmd, args...)
+		execCmd.Stdout = &stdoutBuf
+		execCmd.Stderr = &stderrBuf
+		if err := execCmd.Run(); err != nil {
+			return nil, xerrors.Errorf("get container user: run whoami: %w: stderr: %q", err, strings.TrimSpace(stderrBuf.String()))
+		}
```

Comment:
This is quite a lot of setup, and buf resets can be missed. How about a utility function that handles all of this wiring up? (Including buf resets.). Maybe `out, err := run("whoami")`.

Reply:
Can do! 👍